app/scripts/directives/editWebhookTriggers.js

```diff
@@ -15,18 +16,46 @@ angular.module('openshiftConsole')
       },
       templateUrl: 'views/directives/edit-webhook-triggers.html',
       controller: function($scope) {
+
+        function getWebhookType(type) {
+          switch (type) {
+            case "Generic":
+              return "generic";
+            case "GitHub":
+              return "github";
+            case "GitLab":
+              return "gitlab";
+            case "Bitbucket":
+              return "bitbucket";
+            default:
+              return null;
+          }
+        }
+        $scope.triggerMeta = { 
+          typeToCreate: "",
+          webhookTypes: ["GitHub", "GitLab", "Bitbucket", "Generic"]
+        };
+        $scope.webhookType = getWebhookType($scope.type);
+        if ($scope.webhookType) {
+          $scope.webhookTypeName = $scope.webhookType + "Webhooks";
+        }
+        
         $scope.addWebhookTrigger = function(type) {
+          if (!type) {
+            return;
+          }
           var webhook = {
             disabled: false,
             data: {
               type: type
             }
           };
-          webhook.data[(type === "GitHub") ? "github" : "generic"] = {
+          var webhookType;
+          webhookType = getWebhookType($scope.triggerMeta.typeToCreate);
+          webhook.data[webhookType] = {
             secret: ApplicationGenerator._generateSecret()
           };
-          $scope.triggers.push(webhook);
-          $scope.form.$setDirty();
+          $scope.triggers[webhookType + "Webhooks"].push(webhook);
```

Comment:
But we have to pass the entire map cause we dont know what kind of webhook will user choose. When he chooses GitHub and creates it, we have to add the new webhook to the `githubwebhooks` array, or `bitbucketwebhooks` if he chooses Bitbucket. For that case we need to pass the entire map, just for the creating webhook action. Other option would be to split the create and view action for the webhooks in the `edit-webhook-triggers` and have separate directives for them. Thoughts ?